tools/clang-format/clang-format-diff.py

```diff
@@ -103,13 +107,14 @@ def main():
       if line_count == 0:
         continue
       end_line = start_line + line_count - 1
-      lines_by_file.setdefault((filename, tmpfile), []).extend(
+      lines_by_file.setdefault((filename, tmpfile, filecontent), []).extend(
           ['-lines', str(start_line) + ':' + str(end_line)])
 
   # Reformat files containing changes in place.
-  for filename_dir, lines in lines_by_file.items():
-    filename = filename_dir[0]
-    tmpfile = filename_dir[1]
+  for file_tuple, lines in lines_by_file.items():
```

Comment:
style note: you can write
```for (filename, tmpfile, code), lines in lines_by_file.items():```

Reply:
Ok thx.

Reply:
Regarding "then you want both side to be filename ?": I don't understand

Reply:
> Regarding "then you want both side to be filename ?": I don't understand

Your answer is in a strange place/thread.
If the 2 fields in the difflib.unified_diff() call are only used to *display* the filenames I don't really want any '/tmp/badToto.tmp' , both of them should be the real filename ?

Reply:
Ok.